enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,358 @@
+
+#### What kinds of roles exist related to managedClusterSets and managedClusters
+In General, there are three kinds of roles related to managedClusterSet.
+1. Cluster admin
+   - Cluster admin could `create`, `get`, `list`, `update`, `delete` all managedClusterSets.
+   - Cluster admin could `bind` all managedClusterSets to all namespaces.
+   - Cluster admin could `create`, `get`, `list`, `update`, `delete` all managedClusters.
+   - Only Cluster admin could `accept` managedClusters.
+
+2. Controller/agent which run in managedClusters
+   - The Controller/agent should not `create`, `get`, `list`, `update`, `delete` managedClusterSets.
+   - The Controller/agent should not `bind` managedClusterSet.
+   - The Controller/agent only should `get` related managedCluster and `update` the managedCluster status and labels.
+
+3. Non Cluster admin users
+   - These users should `get` some managedClusterSets, and could not `create`, `update`, `delete`, `list` managedClusterSets.
+   - These users should `bind` some managedClusterSets to their namespaces.
+   - These users may need to `create` managedClusters(For Resource Group Purpose managedClusterSet[d], in each group, the group members may need to create the managedClusters). They could also `delete` these managedClusters, and `update` these managedClusters labels.
+
+#### Who can add labels for managedClusters
+1. Cluster admin
+Cluster admin could add any labels to any managedClusters.
+
+2. Controller/agent which run in managedClusters
+Currentlly, the controller/agent may add labels like: `region: apac`, `cloud: Amazon`, `middlewareEnabled: true`. And these labels mean these clusters have some attributes.
+- `region: apac` means the cluster is provisioned in apac region
+- `cloud: Amazon` means the cluster is provisioned in Amazon cloud provider
+- `middlewareEnabled: true` means this cluster enables the middleware.
+
+3. Non Cluster admin users
+Currently, non cluster admin users may also `add/update` some managedClusters labels. 
+
+#### Add restrictions for adding/updating/deleting labels to managedClusters
+In this proposal, managedClusterSet could use any labels to select managedClusters. And the new added labels may lead to this managedClusters added to a managedClusterSet.
+
+But for some managedClusterSet(like: Disaster Recovery ManagedClusterSet[a], Resource Group ManagedClusterSet[d]), we do not want any users could add their managedClusters to these managedClusterSets. 
+
+So we propose adding restrictions for some labels, and if user want to add these kinds of labels, he/she must have a "special" permission. Then the managedClusterSet which do not want anyone to join could use these labels to select managedClusters.
+
+We propose adding restrictions for labels which key have prefix `cluster.open-cluster-management.io/`. (This prefix may be configured. But in this proposal, it is const)
```

Comment:
I suggest adding another prefix that the poke agents themselves are encouraged to use and have permissions on their own managedcluster to modify. this will give a "trust labels under this prefix to have been set by the managedcluster", which is likely useful for certain kinds of label selection.  This would be comparable to things like the kubelet architecture labels.

Reply:
+1

Reply:
I investigate the kubelet [doc](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) add another prefix `info.open-cluster-management.io/` for spoke agent.